xbmc/listproviders/DirectoryProvider.cpp

```diff
@@ -515,92 +521,111 @@ class CVideoSelectActionProcessor : public VIDEO::GUILIB::CVideoSelectActionProc
   }
 
 private:
-  void BuildAndExecAction(const std::string& method, const std::string& param)
-  {
-    std::vector<std::string> params{StringUtils::Paramify(m_item.GetPath())};
-    if (m_item.m_bIsFolder)
-      params.emplace_back("isdir");
+  CDirectoryProvider& m_provider;
+};
 
-    if (!param.empty())
-      params.emplace_back(param);
+class CVideoPlayActionProcessor : public VIDEO::GUILIB::CVideoPlayActionProcessorBase
+{
+public:
+  explicit CVideoPlayActionProcessor(CFileItem& item) : CVideoPlayActionProcessorBase(item) {}
 
-    const CExecString es{method, params};
-    ExecuteAction(es.GetExecString());
+protected:
+  bool OnResumeSelected() override
+  {
+    ExecuteAction({"PlayMedia", m_item, "resume"});
+    return true;
   }
 
-  CDirectoryProvider& m_provider;
+  bool OnPlaySelected() override
+  {
+    ExecuteAction({"PlayMedia", m_item, "noresume"});
+    return true;
+  }
 };
 } // namespace
 
 bool CDirectoryProvider::OnClick(const CGUIListItemPtr& item)
 {
-  CFileItem fileItem(*std::static_pointer_cast<CFileItem>(item));
+  CFileItem targetItem{*std::static_pointer_cast<CFileItem>(item)};
 
-  if (fileItem.IsFavourite())
+  bool isNoFolderOrIsPlayMedia{false};
+
+  if (targetItem.IsFavourite())
   {
     // Resolve the favourite
-    const CFavouritesURL url{fileItem.GetPath()};
+    const CFavouritesURL url{targetItem.GetPath()};
     if (!url.IsValid())
       return false;
 
-    if (url.GetAction() == CFavouritesURL::Action::PLAY_MEDIA)
+    targetItem = {url.GetTarget(), url.IsDir()};
+    targetItem.LoadDetails();
+
+    isNoFolderOrIsPlayMedia =
```

Comment:
In this particular case I think the grouping makes sense but the naming can be probably improved. Can you name the bool around what the expected logical behaviour is (code flow) instead of on the properties of the item/action? Something like "shouldDoSomething" instead of "isAorB"?

Reply:
I see your point and will improve this, yes.